src/shared/port-finder.js

```diff
@@ -89,11 +78,14 @@ export class PortFinder {
       const listenerId = this._listeners.add(window, 'message', event => {
         const { data, ports } = /** @type {MessageEvent} */ (event);
         if (
-          isMessageEqual(data, {
-            frame1: this._source,
-            frame2: target,
-            type: 'offer',
-          })
+          isMessageEqual(
+            data,
+            /** @type {Message} */ ({
```

Comment:
It is a pity that TS is not able to understand that the combination of `frame1` and `frame2` is a valid combination in `Message`.